src/weights.jl

```diff
@@ -473,116 +473,50 @@ _mean(A::AbstractArray{T}, w::AbstractWeights{W}, dims::Nothing) where {T<:Numbe
-If none of the weights are positive, an error is thrown.
-`NaN` is returned if `x` contains any `NaN` values.
-An error is raised if `w` contains any `NaN` values.
-"""
-function median(v::RealVector, w::AbstractWeights{<:Real})
-    isempty(v) && error("median of an empty array is undefined")
-    if length(v) != length(w)
-        error("data and weight vectors must be the same size")
-    end
-    @inbounds for x in w.values
-        isnan(x) && error("weight vector cannot contain NaN entries")
-    end
-    @inbounds for x in v
-        isnan(x) && return x
-    end
-    mask = w.values .!= 0
-    if !any(mask)
-        error("all weights are zero")
-    end
-    if all(w.values .<= 0)
-        error("no positive weights found")
-    end
-    v = v[mask]
-    wt = w[mask]
-    midpoint = w.sum / 2
-    maxval, maxind = findmax(wt)
-    if maxval > midpoint
-        v[maxind]
-    else
-        permute = sortperm(v)
-        cumulative_weight = zero(eltype(wt))
-        i = 0
-        for (_i, p) in enumerate(permute)
-            i = _i
-            if cumulative_weight == midpoint
-                i += 1
-                break
-            elseif cumulative_weight > midpoint
-                cumulative_weight -= wt[p]
-                break
-            end
-            cumulative_weight += wt[p]
-        end
-        if cumulative_weight == midpoint
-            middle(v[permute[i-2]], v[permute[i-1]])
-        else
-            middle(v[permute[i-1]])
-        end
-    end
-end
-
-
-"""
-    wmedian(v, w)
-
-Compute the weighted median of an array `v` with weights `w`, given as either a
-vector or an `AbstractWeights` vector.
-"""
-wmedian(v::RealVector, w::RealVector) = median(v, weights(w))
-wmedian(v::RealVector, w::AbstractWeights{<:Real}) = median(v, w)
 
 ###### Weighted quantile #####
-
-
 """
     quantile(v, w::AbstractWeights, p)
 
 Compute the weighted quantiles of a vector `v` at a specified set of probability
 values `p`, using weights given by a weight vector `w` (of type `AbstractWeights`).
 Weights must not be negative. The weights and data vectors must have the same length.
-
-With [`FrequencyWeights`](@ref), the function returns the same result as
-`quantile` for a vector with repeated values.
-With non `FrequencyWeights`,  denote ``N`` the length of the vector, ``w`` the vector of weights,
-``h = p (\\sum_{i<= N}w_i - w_1) + w_1`` the cumulative weight corresponding to the
-probability ``p`` and ``S_k = \\sum_{i<=k}w_i`` the cumulative weight for each
+`NaN` is returned if `x` contains any `NaN` values. An error is raised if `w` contains 
+any `NaN` values.
+
+With [`FrequencyWeights`](@ref), the function returns the same result as the unweighted 
+`quantile` applied to a vector with repeated values. The function returns an error if the 
+weights are not of type `AbstractVector{<:Integer}`.
+With non `FrequencyWeights`, denote ``N`` the length of the vector, ``w`` the vector of 
```

Comment:
This paragraph shouldn't be changed (rebasing mistake?).

Reply:
No, it's correct. I added some docs that was in `median` about the behavior with NaN + error about frequency weights that are not integer

Reply:
The changes you mention are above. If you change the wrapping here and remove one sentence that my PR https://github.com/JuliaStats/StatsBase.jl/pull/458 also removes, it won't apply cleanly anymore.

Reply:
I tried to undo some of the change. Does that work for you?

Reply:
I don't understand why it gives conflicts.

Reply:
Ok I went over some git commands I barely understand and it seems to work now.